### PR TITLE
192: Code 403 if logging with incorrect password

### PR DESCRIPTION
**Pull request by IgorMaksymov**
connect to #192

### CODE REVIEW

**Comment by vaa25** [comment]
@IgorMaksymov Why 403?

**Comment by IgorMaksymov** [comment]
@vaa25 trying to acces restricted area.
or its better to 401? or 400?

**Comment by vaa25** [comment]
@IgorMaksymov I think 404 is better, the same as with wrong otp

**Comment by IgorMaksymov** [comment]
@vaa25 broken link is not incorrect login

**Comment by IgorMaksymov** [comment]
@vaa25 i think, that we need to throw EntityNotFound exception if login/password incorrect, not telling what exactly is incorrect.

**Comment by IgorMaksymov** [comment]
@vaa25 should we remove InvalidPasswordException and change it to EntityNotFound with message "user or password incorrect"?

**Comment by vaa25** [comment]
@IgorMaksymov do as @alex-anakin said at tonight meeting. One exception for both login and password fails.

**Comment by IgorMaksymov** [comment]
@vaa25 @alex-anakin said that 401 is correct code

**Comment by romach** [comment]
@IgorMaksymov need to merge `master`

File: restservice/src/test/java/org/itevents/integration/ControllerHandlerTest.java [hunk 1]
```diff
@@ -182,4 +179,22 @@ public void shouldExpect404IfOtpNotValid() throws Exception {
         mvc.perform(get("/users/activate/"+ otp))
                 .andExpect(status().isNotFound());
     }
+
+    @Test
+    public void shouldExpect401WhenLoggingWithIncorrectPassword() throws Exception {
```
Comment (romach) [inline restservice/src/test/java/org/itevents/integration/ControllerHandlerTest.java]: Better name for method: `shouldThrow401WhenLoginWithIncorrectPassword()`

**Comment by IgorMaksymov** [comment]
@romach fixed, but this branch has no conflicts with master, it is faster and easier, than waiting for me.

File: restservice/src/test/java/org/itevents/integration/ControllerHandlerTest.java (hunk 1 above)
Comment (IgorMaksymov) [inline restservice/src/test/java/org/itevents/integration/ControllerHandlerTest.java]: throwing is for exceptions, we are expecting HTTP code 401

**Comment by vkuchyn** [comment]
@IgorMaksymov it is best practices before PR merge master into feature branch. It will prevent from lots of problems in future. It is responsibility of developer, not reviewer. So, for future - please keep in mind this point

**Comment by IgorMaksymov** [comment]
@vkuchyn yes, i merge my branch with master before PR. Idea was to merge master into feature branch by reviewer if it has no conflict with master. If there is some conflicts - author solves them by himself.

**Comment by vkuchyn** [comment]
@IgorMaksymov I am not agree with your point. Only author of the branch is responsible to do some changes. If master was updated when issue was in review and no conflicts - ok, could happen. But keep in mind - every time you do some changes to branch - check if there are no any updates from master and merge them if so

**Comment by IgorMaksymov** [comment]
@vkuchyn ok, i will merge master even if there is no conflicts before making new commits to PR.

**Comment by romach** [comment]
@AndriyBaibak 30 minutes spent

**Pull request merged**